### PR TITLE
[MAINTENANCE] Lint `test/data_context`

### PR DESCRIPTION
**Pull request by Kilo59**
### Changes proposed in this pull request:
- begin linting `tests/data_context` - remove from `pyproject.toml` `ruff` exclude
- apply auto-fixes 
- add `noqa` comments to remaining warnings

### CODE REVIEW

File: tests/data_context/store/test_store_backends.py [hunk 1]
```diff
@@ -1054,10 +1054,7 @@ def test_TupleGCSStoreBackend_base_public_path():
     project = "dummy-project"
     base_public_path = "http://www.test.com/"
 
-    with mock.patch("google.cloud.storage.Client", autospec=True) as mock_gcs_client:
-        mock_client = mock_gcs_client.return_value
-        mock_bucket = mock_client.get_bucket.return_value
-        mock_blob = mock_bucket.blob.return_value
```
Comment (Kilo59) [inline tests/data_context/store/test_store_backends.py:1060]: None of these were being used 🤷

File: tests/data_context/datasource/test_data_context_datasource_non_sql_methods_sparkdf_execution_engine.py [hunk 2]
```diff
@@ -233,7 +235,7 @@ def test_get_batch_list_from_datasource_with_runtime_data_connector(
         **config,
     )
 
-    test_df: "pyspark.sql.dataframe.DataFrame" = (  # noqa: F821
+    test_df: pyspark.sql.dataframe.DataFrame = (  # noqa: F821
```
Comment (Kilo59) [inline tests/data_context/datasource/test_data_context_datasource_non_sql_methods_sparkdf_execution_engine.py]:
```suggestion
    test_df: pyspark.sql.dataframe.DataFrame = (
```

File: tests/data_context/test_data_context_test_yaml_config.py [hunk 3]
```diff
@@ -422,7 +422,7 @@ def test_datasource_config(mock_emit, caplog, empty_data_context_stats_enabled):
     print(json.dumps(return_obj, indent=2))
 
     assert set(return_obj.keys()) == {"execution_engine", "data_connectors"}
-    sub_obj = return_obj["data_connectors"]["my_filesystem_data_connector"]
```
Comment (Kilo59) [inline tests/data_context/test_data_context_test_yaml_config.py:425]: add back?

File: pyproject.toml [hunk 4]
```diff
@@ -275,7 +275,6 @@ extend-exclude = [
     "tests/checkpoint", # 15 warnings
     "tests/cli", # 34 warnings
     "tests/data_asset", # 10 warnings
-    "tests/data_context", # 127 warnings
```
Comment (Kilo59) [inline pyproject.toml:278]: Removing this from the `ruff` exclude rules

**Review by cdkini — APPROVED** [review]

**Pull request merged**